include/ck_tile/ops/gemm/pipeline/block_gemm_pipeline_problem.hpp

```diff
@@ -5,21 +5,32 @@
 
 #include "ck_tile/core.hpp"
 
+#define VectorLoadSize 16
+
 namespace ck_tile {
 
 template <typename ADataType_,
           typename BDataType_,
           typename CDataType_,
-          index_t kBlockSize_,
-          typename BlockGemmShape_>
+          typename BlockGemmShape_,
+          bool kPadA_ = false,
+          bool kPadB_ = false,
+          bool kPadC_ = false>
 struct BlockGemmPipelineProblem
 {
     using ADataType      = remove_cvref_t<ADataType_>;
     using BDataType      = remove_cvref_t<BDataType_>;
     using CDataType      = remove_cvref_t<CDataType_>;
     using BlockGemmShape = remove_cvref_t<BlockGemmShape_>;
 
-    static constexpr index_t kBlockSize = kBlockSize_;
+    static constexpr index_t kBlockSize = BlockGemmShape::NumWarps * get_warp_size();
```

Comment:
`kBlockSize` is a confusing name. People may think it is `kPerBlock`. Actually, all `k` prefix is confusing.

Reply:
Because in the do-while loop it if we only have the num_loop = 1 and the iCounter is 0. It will execute this block of the code. However, it should skip that and go to the tile block directly.

Reply:
Changed the variable name. For the other k-headed variables as there are too many that correlates with the FMHA and inside GEMM block, I will reformat all of the variable names in the next pull request.